pkg/source/source.go

```diff
@@ -14,7 +14,7 @@ type Source string
 
 const (
 	// Unspec is used when the source is unspecified
-	Unspec Source = "unspec"
+	Unspec Source = ""
```

Comment:
Why does the updated node have an empty string as source? Isn't that also a bug, or is that *only* in the test and the problem is that the test is not calling the API with a proper source in the way that the production code would?

Reply:
`Unspec` is the default value returned by `func (s prefixInfo) Source() source.Source` in `pkg/ipcache/types.go`.

I'm not sure if `TestNodeLifecycle` verifies a realistic or artificial scenario, but the change here unifies two unspecified values `source.Unspec` and `"" string`, which seems to make sense (unless https://github.com/cilium/cilium/blob/b424bfda4f686fb34271c91d292372f4353477b2/pkg/ipcache/types.go#L39-L42 requires two unspecified values).

Reply:
I checked calls to `func (m *manager) NodeUpdated(n nodeTypes.Node)`:

- The call from `pkg/k8s/watchers/cilium_node.go` has source specified.
- The call from `pkg/node/store/store.go` as well.
-  I'm not sure about the calls from `StartDiscovery()` in `pkg/nodediscovery/nodediscovery.go`. These calls use as the argument the `Node` field of `LocalNode` object. Now, there are two non-test places where the `Node` field of `LocalNode` are set. One (`parseNode()` in `daemon/cmd/local_node_sync.go`) sets the source and the other (`NewLocalNodeStore()` in `pkg/node/local_node_store.go`) does not. Not sure if both of these places are relevant to the calls in `StartDiscovery()`.

Reply:
Oh, and specifying an explicit source in `TestNodeLifecycle` makes it work without changing the value of `Unspec` to an empty string. So I'm also fine with modifying `TestNodeLifecycle` and keeping `Unspec` as it was.

Could you please let me know what is the best approach here?

Reply:
I've isolated these changes to https://github.com/cilium/cilium/pull/36095, we can continue the discussion there.